### PR TITLE
Add Terraform Infra for `knative-dns` and `knative-gsuite` projects

### PR DESCRIPTION
**Pull request by upodroid**
~Work In Progress~
- Need to create prowjobs
- ~add tfenv to prow-tests image~
- pending projects
  - knative-performance, knative-eventing-performance
  - knative-gcp
  - knative-releases, knative-nightly
  - knative-tests

/cc @chizhg @dprotaso @kvmware

**Which issue(s) this PR fixes**:<br>

### CODE REVIEW

**Comment by kvmware** [comment]
Please remove the hold once it is out of WIP

**Comment by upodroid** [comment]
/hold cancel

Happy to merge this. Will log separate PRs for other set of projects.

File: infra/gcp/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# Get the ORG_ID
+ORG_ID=$(gcloud organizations describe knative.team --format json | jq .name -r | sed 's:.*/::')
+
+#Create the knative-seed project
+
+gcloud projects create knative-seed --organization $ORG_ID --name "Knative Seed"
+
+#Create the terraform service account
+
+gcloud iam service-accounts create terraform —-display-name “Terraform” --project knative-seed
```
Comment (chizhg) [inline infra/gcp/README.md:34]: What IAM bindings do we need for this service account? 
Also, is this one-off, or do we still need this service account and `knatve-seed` project for future operations?

**Review by chizhg — COMMENTED** [review]
I'm not familiar with Terraform. One quick question, what will be the command to apply these changes? Will it be easy to setup automation for it?

File: infra/gcp/README.md (hunk 1 above)
Comment (upodroid) [reply infra/gcp/README.md:34]: It is already been added.

```
 REDACTED  MCW0CDP3YY  ~  Desktop  Git  $  gcloud organizations get-iam-policy 22054930418
bindings:
- members:
  - domain:knative.team
  role: roles/billing.creator
- members:
  - serviceAccount:service-org-22054930418@security-center-api.iam.gserviceaccount.com
  role: roles/cloudfunctions.serviceAgent
- members:
  - domain:knative.team
  - group:kn-infra-gcp-org-admins@knative.team
  role: roles/cloudsupport.techSupportEditor
- members:
  - group:kn-infra-gcp-org-admins@knative.team
  role: roles/logging.privateLogViewer
- members:
  - domain:knative.team
  role: roles/logging.viewer
- members:
  - group:kn-infra-gcp-org-admins@knative.team
  - serviceAccount:terraform@knative-seed.iam.gserviceaccount.com
  role: roles/orgpolicy.policyAdmin
- members:
  - group:kn-infra-gcp-org-admins@knative.team
  - serviceAccount:terraform@knative-seed.iam.gserviceaccount.com
  role: roles/owner
- members:
  - group:kn-infra-gcp-org-admins@knative.team
  - serviceAccount:terraform@knative-seed.iam.gserviceaccount.com
  role: roles/resourcemanager.folderAdmin
- members:
  - group:kn-infra-gcp-org-admins@knative.team
  - serviceAccount:terraform@knative-seed.iam.gserviceaccount.com
  role: roles/resourcemanager.organizationAdmin
- members:
  - domain:knative.team
  - group:kn-infra-gcp-org-admins@knative.team
  - serviceAccount:terraform@knative-seed.iam.gserviceaccount.com
  role: roles/resourcemanager.projectCreator
- members:
  - serviceAccount:service-org-22054930418@security-center-api.iam.gserviceaccount.com
  role: roles/securitycenter.serviceAgent
- members:
  - serviceAccount:service-org-22054930418@security-center-api.iam.gserviceaccount.com
  role: roles/serviceusage.serviceUsageAdmin
etag: BwXbiEECEGA=
version: 1
```

We still need the service account as that is the account prow will use to do terraform changes. Humans will only need to run terraform if prow is unavailable.

knative-seed project holds the service account and the terraform state bucket

**Comment by upodroid** [comment]
> Oh also another thing, as we are now moving the infra stuff to the github.com/knative/infra repo, would it make more sense to directly create this PR there?

It is not ready yet.

**Comment by upodroid** [comment]
Can I get lgtm for this? Will raise separate PR for each project

**Pull request merged**